src/interpreter_funcs/voxel_boolean_union.rs

```diff
@@ -218,30 +221,77 @@ impl Func for FuncBooleanUnion {
-        let mut voxel_cloud2 = ScalarField::from_mesh(mesh2, &voxel_dimensions, 0.0, growth_u32);
+        // FIXME: The growth + 1 here is only because it seems like some
+        // chopping happens. This is one place where the chopping may occur.
+        let mut voxel_cloud1 =
+            ScalarField::from_mesh(mesh1, &voxel_dimensions, 0.0, growth_u32 + 1);
+        let mut voxel_cloud2 =
+            ScalarField::from_mesh(mesh2, &voxel_dimensions, 0.0, growth_u32 + 1);
 
-        voxel_cloud1.compute_distance_field(&(0.0..=0.0), FalloffFunction::Linear(1.0));
-        voxel_cloud2.compute_distance_field(&(0.0..=0.0), FalloffFunction::Linear(1.0));
+        let original_volume_value_range = 0.0..=0.0;
 
-        let meshing_range = if fill {
-            (Bound::Unbounded, Bound::Included(growth_f32))
-        } else {
-            (Bound::Included(-growth_f32), Bound::Included(growth_f32))
-        };
+        voxel_cloud1.compute_distance_field(
+            &original_volume_value_range,
+            1.0,
+            FalloffFunction::Linear,
+        );
+        voxel_cloud2.compute_distance_field(
+            &original_volume_value_range,
+            1.0,
+            FalloffFunction::Linear,
+        );
+
+        if fill {
+            let mut voxel_cloud1_inside =
+                voxel_cloud1.extract_voxels_inside_volumes(&original_volume_value_range);
+
+            let mut voxel_cloud2_inside =
+                voxel_cloud2.extract_voxels_inside_volumes(&original_volume_value_range);
+
+            voxel_cloud1_inside.apply_unary_voxel_function(UnaryVoxelFunction::SetConstant(0.0));
+            voxel_cloud2_inside.apply_unary_voxel_function(UnaryVoxelFunction::SetConstant(0.0));
 
-        voxel_cloud1.boolean_union(&meshing_range, &voxel_cloud2, &meshing_range);
+            voxel_cloud1.apply_binary_voxel_function(
+                &voxel_cloud1_inside,
+                BinaryVoxelFunction::ReplaceIfValue,
+            );
+            voxel_cloud2.apply_binary_voxel_function(
+                &voxel_cloud2_inside,
+                BinaryVoxelFunction::ReplaceIfValue,
+            );
+        }
+
+        let bounding_box_voxel_cloud1 = voxel_cloud1.bounding_box_cartesian_space();
+        let bounding_box_voxel_cloud2 = voxel_cloud2.bounding_box_cartesian_space();
+
+        let volume_value_range = (Bound::Included(-growth_f32), Bound::Included(growth_f32));
+
+        // FIXME: Visually it seems like this is chopping off some parts of the
+        // volume. This is another place where the chopping may occur.
+        if let Some(bounding_box) = BoundingBox::union(
+            [bounding_box_voxel_cloud1, bounding_box_voxel_cloud2]
+                .iter()
+                .copied(),
+        ) {
+            voxel_cloud1.resize_to_bounding_box_cartesian_space(&bounding_box);
+            voxel_cloud2.resize_to_bounding_box_cartesian_space(&bounding_box);
```

Comment:
If I understand correctly, we don't need to resize the second scalar fields. Values outside will just be void.